service/graph.go

```diff
@@ -356,3 +356,36 @@ func (p *pipelineNodes) exporterIDs() []string {
 func (p *pipelineNodes) mutatesData() bool {
 	return p.capabilitiesNode.getConsumer().Capabilities().MutatesData
 }
+
+func cycleErr(err error) error {
+	var topoErr topo.Unorderable
+	if !errors.As(err, &topoErr) {
+		return err
+	}
+
+	// There may be multiple cycles, but report only the first one.
```

Comment:
Why only the first one?

Reply:
It's just an attempt to limit the size of the error message.